### PR TITLE
Feature/save variables

### PR DESCRIPTION
**Pull request by cfrank**
Takes the key and value strings from #14 and saves them into a `config_value` which now needs to be saved into the `config_parser` `config_list`

### CODE REVIEW

File: src/common/util.c [hunk 1]
```diff
@@ -189,6 +192,48 @@ ssize_t string_get_delimiter(const char *string, char delimiter,
...
+
+        intmax_t result = strtoimax(number_string, &endptr, 10);
+
+        if (errno == ERANGE) {
+                switch (result) {
+                case INTMAX_MIN:
+                case INTMAX_MAX:
+                        return -1;
+                default:
+                        LOG_CRITICAL_LONG(
```
Comment (cfrank) [inline src/common/util.c]: Use `goto` to clean up these two logs

File: src/core/config.c [hunk 2]
```diff
@@ -423,6 +484,27 @@ static struct config_list *parse_file(FILE *file)
...
+        for (size_t i = 0; i < list->length; ++i) {
+                destroy_config_value(list->values[i]);
+        }
+
+        free(list);
+}
+
+void destroy_config_value(struct config_value *value)
+{
+        if (value->type == NUMBER) {
```
Comment (cfrank) [inline src/core/config.c]: swap this conditional to look for `value->type == STRING` and just free the value to avoid free'ing `value` twice

File: src/test/test_string_util.c [hunk 3]
```diff
@@ -249,6 +249,54 @@ static void test_string_get_delimiter_empty_string(void **state)
         assert_int_equal(expected_result, result);
 }
 
+static void test_string_to_number(void **state)
```
Comment (cfrank) [inline src/test/test_string_util.c:252]: Add test for empty string
